deployer/src/deployer/main.py

```diff
@@ -1,101 +1,99 @@
-import functools
-import pkg_resources
 from pathlib import Path
 
 import click
 
+from . import __version__
+from .aws_lambda import update_all as update_all_lambda_functions
 from .constants import (
-    DEFAULT_NAME,
-    DEFAULT_NAME_PATTERN,
-    DEFAULT_BUCKET,
-    S3_DEFAULT_BUCKET_LOCATION,
+    CONTENT_ROOT,
+    CONTENT_TRANSLATED_ROOT,
+    DEFAULT_BUCKET_NAME,
     DEFAULT_NO_PROGRESSBAR,
 )
-from .exceptions import CoreException
-from .upload import upload_site
-from .utils import error, info
-from . import __version__
-
-
-def cli_wrap(fn):
-    @functools.wraps(fn)
-    def inner(*args, **kwargs):
-        try:
-            fn(*args, **kwargs)
-        except CoreException as exception:
-            info(exception.__class__.__name__)
-            error(str(exception))
-            raise click.Abort
-
-    return inner
+from .upload import upload_content
+from .utils import log
 
 
 @click.group()
-@click.option("--debug/--no-debug", default=False)
+@click.option(
+    "--dry-run",
+    default=False,
+    help="Show what would be done, but don't actually do it.",
+    show_default=True,
+    is_flag=True,
+)
 @click.version_option(version=__version__)
 @click.pass_context
-def cli(ctx, debug):
+def cli(ctx, **kwargs):
     ctx.ensure_object(dict)
-    ctx.obj["debug"] = debug
+    ctx.obj.update(kwargs)
 
 
 @cli.command()
 @click.pass_context
-@cli_wrap
+def update_lambda_functions(ctx):
+    log.info(f"Deployer ({__version__})", bold=True)
+    update_all_lambda_functions(dry_run=ctx.obj["dry_run"])
+
+
+@cli.command()
 @click.option(
     "--bucket",
-    default=DEFAULT_BUCKET,
-    help=f"Name of the bucket (default {DEFAULT_BUCKET!r})",
-)
-@click.option(
-    "--name",
-    default=DEFAULT_NAME,
-    help=f"Name of the site (default {DEFAULT_NAME_PATTERN!r})",
+    help='Name of the S3 bucket or one of "dev", "stage", or "prod"',
+    default=DEFAULT_BUCKET_NAME,
+    show_default=True,
 )
 @click.option(
-    "--bucket-location",
-    default=S3_DEFAULT_BUCKET_LOCATION,
-    help=(
-        f"Name of the S3 bucket location (like 'us-east-1') "
-        f"(default {S3_DEFAULT_BUCKET_LOCATION!r})"
-    ),
+    "--folder", help="Upload into this folder of the S3 bucket instead of its root",
```

Comment:
The `--folder` option only makes sense for use with the `dev` bucket, since the dev environment is the only environment for which we'll prefix the key request to S3 with the first part of the domain name (e.g., `pr1234.content.dev.mdn.mozit.cloud`). I'll change the name to `prefix`.

Reply:
I think we should always have a prefix. Even for the production site. That way we're consistent. 
Suppose we ignore "stage" for a minute. What I had in mind was:

* Bucket: mdn-content-prod
  - prefix: main/
  - prefix: schalks-new-redesign/
  - prefix: experiment-inline-bcd-table/
* Bucket: mdn-content-pr-builds
  - prefix: pr123/
  - prefix: pr124/
  - prefix: pr126/
  - prefix: pr...

(`main` instead of `master`)

With a prefix, you can re-use the bucket for multiple sites.

Reply:
We discussed this. We're arguing about exact same thing just two equivalent ways of doing the same thing. 
So, fine like this.